python/tests/test_boublik_hard_sphere.py

```diff
@@ -50,7 +51,7 @@ def test_compute(bmcsl,binary_state):
 
     # compute with only one component present, other acts like ideal gas
     bmcsl.compute(state)
-    assert bmcsl.value == pytest.approx(10.*fex_cs(eta,d))
+    assert bmcsl.value == pytest.approx(volumes*fex_cs(eta,d))
```

Comment:
Just call the volume method:
```suggestion
    assert bmcsl.value == pytest.approx(state.mesh.full.volume()*fex_cs(eta,d))
```